lib/src/facts/posix/networking_resolver.cc

```diff
@@ -106,7 +110,7 @@ namespace facter { namespace facts { namespace posix {
                     result.domain.assign(parts[1].begin(), parts[1].end());
                     return false;
                 }
-                if (parts[0] == boost::as_literal("search")) {
+                if (search.empty() && parts[0] == boost::as_literal("search")) {
```

Comment:
Since I have to push up an amended commit anyway, let me pull this out into its own commit.